mention renamings on `WSRequest` in 2.6 migration

mriehl opened the pull request:
This addresses [play-ws#191](https://github.com/playframework/play-ws/issues/191).

It's worth mentioning because the nearest match for `withHeaders` doesn't have the same behaviour.

# Pull Request Checklist

* [x] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
* [x] Have you read through the [contributor guidelines](https://www.playframework.com/contributing)?
* [x] Have you signed the [Lightbend CLA](https://www.lightbend.com/contribute/cla)?
* [x] Have you [squashed your commits]? (Optional, but makes merge messages nicer.)
* [x] Have you added copyright headers to new files?
* [x] Have you checked that both Scala and Java APIs are updated?
* [x] Have you updated the documentation for both Scala and Java sections?
* [x] Have you added tests for any changed functionality?

# Helpful things

## Fixes

Fixes [play-ws#191](https://github.com/playframework/play-ws/issues/191)

## Purpose

See [play-ws#191](https://github.com/playframework/play-ws/issues/191)

## Background Context

See [play-ws#191](https://github.com/playframework/play-ws/issues/191)

## References

See [play-ws#191](https://github.com/playframework/play-ws/issues/191)

marcospereira commented on documentation/manual/releases/release26/migration26/WSMigration26.md:
```diff
@@ -81,6 +81,10 @@ The `ning` package has been replaced by the `ahc` package, and the Ning* classes
 A normal `WSResponse` instance is returned from `stream()` instead of `StreamedResponse`.  You should call `response.bodyAsSource` to return the streamed result.
 
+There are some naming changes with deprecations to make things more explicit on `play.api.libs.ws.WSRequest`. Extra care should be exercised when migrating these:
+- `wsRequest.withHeaders` is now `wsRequest.addHttpHeaders` (same behaviour) or `wsRequest.withHttpHeaders` (throws away existing headers)
```
Can you add links to scaladoc here? It would be something like:

```
- [`WsRequest.withHeaders`](https://playframework.com/documentation/2.6.x/api/scala/index.html#play.api.libs.ws.WSRequest@withHeaders\(headers:\(String,String\)*\):WSRequest.this.Self) is now [`WsRequest.addHttpHeaders`](https://playframework.com/documentation/2.6.x/api/scala/index.html#play.api.libs.ws.WSRequest@addHttpHeaders\(headers:\(String,String\)*\):WSRequest.this.Self) (same behaviour) or [`WsRequest.withHttpHeaders`](https://playframework.com/documentation/2.6.x/api/scala/index.html#play.api.libs.ws.WSRequest@withHttpHeaders\(headers:\(String,String\)*\):WSRequest.this.Self) (throws away existing headers)
```
And the same for the next bullet point.

mriehl commented:
@marcospereira Thanks for your review. I didn't know it was possible to do deeplinks to the scaladoc :-/

Both added as per your request. I squashed and rebased for clarity.

marcospereira commented:
Backport to 2.6.x here: 01be59e355b2f864c80518416d89b8711e1bc46e

Thank you, @mriehl!
